pkg/sql/parsers/dialect/mysql/mysql_sql.y

```diff
@@ -2107,7 +2107,13 @@ show_stmt:
 |   show_target_filter_stmt
 |   show_table_status_stmt
 |   show_grants_stmt
+|   show_collation_stmt
 
+show_collation_stmt:
+    SHOW COLLATION like_opt where_expression_opt
+    {
+        $$ = &tree.ShowCollation{}  
```

Comment:
may be you lost your like/ where field?

Reply:
and you can add ut in  mysql_sql_test.go

Reply:
> and you can add ut in mysql_sql_test.go

fixed